.github/workflows/update-libs.yaml

```diff
@@ -0,0 +1,50 @@
+name: Auto-update Charm Libraries
+on:
+  workflow_call:
+    secrets:
+      CHARMHUB_TOKEN:
+        required: false
+      OBSERVABILITY_NOCTUA_TOKEN:
+        required: true
+
+permissions:
+  contents: write
+  pull-requests: write
+
+jobs:
+  update-lib:
+    name: Check libraries
+    runs-on: ubuntu-22.04
+    steps:
+      - name: Checkout
+        uses: actions/checkout@ac593985615ec2ede58e132d2e21d2b1cbd6127c # v3
+        with:
+          fetch-depth: 0
+
+      - name: Check libs
+        run: |
+          sudo snap install charmcraft --classic --channel latest/stable
+          charmcraft fetch-lib
+        env:
+          CHARMCRAFT_AUTH: "${{ secrets.CHARMHUB_TOKEN }}"
+
+      - name: Create a PR for local changes
+        uses: peter-evans/create-pull-request@v4.2.3
+        id: cpr
+        with:
+          token: ${{ secrets.OBSERVABILITY_NOCTUA_TOKEN }}
+          commit-message: "chore: bump charm libraries"
+          committer: "Github Actions <github-actions@github.com>"
+          author: "Github Actions <github-actions@github.com>"
+          title: "Bump charm libraries"
+          body: Automated action to fetch latest version of charm libraries.
+          branch: "chore/auto-libs"
+          delete-branch: true
+          reviewers: |
```

Comment:
Our repos already has all of us as default reviewers. Not sure if we need it here too.